react-components/src/architecture/concrete/observationsDomainObject/ObservationsDomainObject.ts

```diff
@@ -1,28 +1,29 @@
 /*!
  * Copyright 2024 Cognite AS
  */
-import { CDF_TO_VIEWER_TRANSFORMATION, type Overlay3D, Overlay3DCollection } from '@cognite/reveal';
-import { OBSERVATION_SOURCE, type ObservationProperties, type Observation } from './models';
+import { type Overlay3D } from '@cognite/reveal';
+import { type ObservationProperties } from './models';
 import { VisualDomainObject } from '../../base/domainObjects/VisualDomainObject';
 import { type ThreeView } from '../../base/views/ThreeView';
 import { ObservationsView } from './ObservationsView';
 import { type TranslateKey } from '../../base/utilities/TranslateKey';
-import { type FdmSDK, type InstanceFilter } from '../../../utilities/FdmSDK';
-import { Vector3 } from 'three';
-import { DEFAULT_OVERLAY_COLOR } from './constants';
+import { type FdmSDK } from '../../../utilities/FdmSDK';
 import { Changes } from '../../base/domainObjectsHelpers/Changes';
+import { ObservationsCache } from './ObservationsCache';
+import { type Vector3 } from 'three';
+import { PanelInfo } from '../../base/domainObjectsHelpers/PanelInfo';
+import { type ObservationCollection, type ObservationOverlay } from './types';
+import { ObservationStatus } from './ObservationStatus';
 
 export class ObservationsDomainObject extends VisualDomainObject {
```

Comment:
I will strongly prefer that the domain object has a collection of Observation, and no references to Overlay3D object. Move the Overlay stuff to the view instead.